### ISSUE TITLE
ERROR - 'tcp://10.33.3.2:42745' when using dask

### ISSUE THREAD

**Issue by hardingnj** [malariagen/datalab#71]
Can't see a report- apologies if this is known, but I get these errors when using dask:

> Traceback (most recent call last):
>   File "/opt/conda/lib/python3.6/site-packages/distributed/utils.py", line 666, in log_errors
>     yield
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 1499, in add_worker
>     await self.handle_worker(comm=comm, worker=address)
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 2408, in handle_worker
>     worker_comm = self.stream_comms[worker]
> KeyError: 'tcp://10.33.15.2:35089'
> distributed.utils - ERROR - 'tcp://10.33.9.3:42121'
> Traceback (most recent call last):
>   File "/opt/conda/lib/python3.6/site-packages/distributed/utils.py", line 666, in log_errors
>     yield
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 1499, in add_worker
>     await self.handle_worker(comm=comm, worker=address)
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 2408, in handle_worker
>     worker_comm = self.stream_comms[worker]
> KeyError: 'tcp://10.33.9.3:42121'
> distributed.utils - ERROR - 'tcp://10.32.254.2:41423'
> Traceback (most recent call last):
>   File "/opt/conda/lib/python3.6/site-packages/distributed/utils.py", line 666, in log_errors
>     yield
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 1499, in add_worker
>     await self.handle_worker(comm=comm, worker=address)
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 2408, in handle_worker
>     worker_comm = self.stream_comms[worker]
> KeyError: 'tcp://10.32.254.2:41423'
> distributed.utils - ERROR - 'tcp://10.33.3.2:42745'
> Traceback (most recent call last):
>   File "/opt/conda/lib/python3.6/site-packages/distributed/utils.py", line 666, in log_errors
>     yield
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 1499, in add_worker
>     await self.handle_worker(comm=comm, worker=address)
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 2408, in handle_worker
>     worker_comm = self.stream_comms[worker]
> KeyError: 'tcp://10.33.3.2:42745'
> distributed.utils - ERROR - 'tcp://10.33.15.3:43249'
> Traceback (most recent call last):
>   File "/opt/conda/lib/python3.6/site-packages/distributed/utils.py", line 666, in log_errors
>     yield
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 1499, in add_worker
>     await self.handle_worker(comm=comm, worker=address)
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 2408, in handle_worker
>     worker_comm = self.stream_comms[worker]
> KeyError: 'tcp://10.33.15.3:43249'
> distributed.utils - ERROR - 'tcp://10.33.11.3:37789'
> Traceback (most recent call last):
>   File "/opt/conda/lib/python3.6/site-packages/distributed/utils.py", line 666, in log_errors
>     yield
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 1499, in add_worker
>     await self.handle_worker(comm=comm, worker=address)
>   File "/opt/conda/lib/python3.6/site-packages/distributed/scheduler.py", line 2408, in handle_worker
>     worker_comm = self.stream_comms[worker]
> KeyError: 'tcp://10.33.11.3:37789'
> distributed.utils - ERROR - 'tcp://10.32.254.4:40659'
> Traceback (most recent call last):

The error doesn't stop the notebook kernel, but still concerning. Is this a known issue or something with our configuration?

Thanks!

**Comment by slejdops** [comment]
I'll look into it

**Comment by hardingnj** [comment]
Removing `cluster.adapt()` seems to fix this issue.

**Comment by slejdops** [comment]
we're  using dask-kubernetes not dask-jobqueue
jobqueue is for deploying dask on SGE, SLURM , etc. 
https://jobqueue.dask.org/en/latest/

**Comment by hardingnj** [comment]
Ah, thanks- I guess the cluster adapt logic is similar though?

**Comment by slejdops** [comment]
Yes, I think the errors you're getting might be related to this 
distributed.nanny - INFO - Closing Nanny at 'tcp://10.32.60.2:35879'
distributed.worker - INFO - Stopping worker at tcp://10.32.60.2:41549
distributed.worker - INFO - **Closed worker has not yet started: None**

that's an error from one of your dask workers
